File: Mod07/Labfiles/Solution/VB/ClassTracker_VB/ClassTracker_VB/Course.vb
Public Class Course

    Private _courseTitle As String
    Private _creditHours As Integer
    Private _program As String

    Public Property CourseTitle As String
        Get
            Return Me._courseTitle
        End Get
        Set(value As String)
            Me._courseTitle = value
        End Set
    End Property

    Public Property CreditHours As Integer
        Get
            Return Me._creditHours
        End Get
        Set(value As Integer)
            Me._creditHours = value
        End Set
    End Property

    Public Property Program As String
        Get
            Return Me._program
        End Get
        Set(value As String)
            Me._program = value
        End Set
    End Property

    Public Sub EnrollStudent(studentName As String)

    End Sub

    Public Overridable Function CalculateAverage(arrGrades() As Single) As Single
        Dim avg As Single = 0.0

        Return avg
    End Function

End Class

File: Mod07/Labfiles/Solution/CS/ClassTracker_CS/ClassTracker_CS/Course.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassTracker_CS
{
    class Course
    {
        private string courseTitle;
        private int creditHours;
        private string program;

        public string CourseTitle
        {
            get { return courseTitle; }
            set { courseTitle = value; }
        }

        public int CreditHours
        {
            get { return creditHours; }
            set { creditHours = value; }
        }

        public string Program
        {
            get { return program; }
            set { program = value; }
        }

        public void EnrollStudent(string studentName)
        {

        }

        public virtual float CalculateAverage(float[] arrGrades)
        {
            float avg = 0;

            return avg;
        }
    }
}
